File: src/main/resources/static/js/service/associationService.js
//服务层
app.service('associationService', function ($http) {

    //审核中心页面，查询所有社团信息
    this.findAll = function () {
        return $http.get('/association/findAll');
    };
    //分页
    this.findPage = function (page, rows) {
        return $http.get('../association/findPage.do?page=' + page + '&rows=' + rows);
    }
    //通过登录网址管理员信息，查询此管理员加入的所有是管理角色的社团
    this.findByStuCodeAndRole = function (stuCode, role) {
        return $http.get('/association/findByStuCodeAndRole/' + stuCode + '/' + role);
    };
    //查询一个社团详情
    this.findOne = function (id) {
        return $http.get('/association/findOne/' + id);
    };
    //通过社团类型查询所有社团
    this.findByTypeCode = function (code) {
        return $http.get('/association/findByTypeCode/' + code);
    };
    //社团申请的service,entity为传输的添加数据
    this.add = function (entity) {
        return $http.post('/association/add', entity);
    };
    //修改社团信息
    this.update = function (entity) {
        return $http.post('/association/update', entity);
    };
    //删除社团信息
    this.dele = function (id, stuCode) {
        return $http.delete('/association/delete/' + id + "/" + stuCode);
    };
    //换届前查询此社团所有学生
    this.findOneAss = function (assId) {
        return $http.get('/association/findOneAss/' + assId)
    };
    //社长换届
    this.changeManage = function (changeManageStuCode, changeManageAssId) {
        return $http.post('/association/changeManage/' + changeManageStuCode + '/' + changeManageAssId)
    };
    //查询所有社团service，搜索条件默认为null
    this.search = function (page, rows, searchEntity) {
        return $http.post('/association/search/' + page + "/" + rows, searchEntity);
    };
});

File: src/main/java/com/ccut/yiyi/service/AssociationService.java
package com.ccut.yiyi.service;

import com.ccut.yiyi.model.Association;
import com.ccut.yiyi.model.Student;
import com.ccut.yiyi.model.group.AssoSimplify;
import com.ccut.yiyi.model.group.AssociationApply;
import com.ccut.yiyi.model.group.AssociationGroup;
import com.ccut.yiyi.model.group.StudentSimplify;
import org.springframework.data.domain.Page;

import java.util.List;
import java.util.Map;

/**
 * @ClassName: AssociationService
 * @Description: TODO
 * @Author: Mark
 * @Date: 2019/12/22 14:47
 * @version: V1.0
 */
public interface AssociationService {
    /**
     * 社团申请service接口
     *
     * @param associationApply
     */
    void add(AssociationApply associationApply);

    /**
     * 社团条件查询接口 默认查询条件为null
     * 条件查询+分页
     *
     * @param whereMap
     * @param page
     * @param size
     * @return
     */
    Page<AssociationGroup> findSearch(Map whereMap, int page, int size);

    /**
     * UI页面查询社团信息
     *
     * @param page 页数
     * @param size 条数
     * @param id   大类id
     * @return 返回数据
     */
    Page<Association> findSearch(int page, int size, int id);

    /**
     * 通过社团类型typeCode查询其下所有社团
     *
     * @param code
     * @return
     */
    List<Association> findAssByCode(Integer code);

    /**
     * 根据社团id查询一个的接口
     *
     * @param id
     * @return
     */
    AssociationGroup findOne(Integer id);

    /**
     * 删除社团信息
     *
     * @param id
     */
    void deleteById(Integer id, String stuCode);

    /**
     * 修改社团信息
     *
     * @param associationGroup
     */
    void update(AssociationGroup associationGroup);

    /**
     * 通过社团id查询此社团下的所有学生
     *
     * @param assId 社团id
     * @return 返回数据
     */
    List<StudentSimplify> findOneAss(Integer assId);

    /**
     * 社团换届的service接口
     *
     * @param stuCode 需要换届的学生学号
     * @param assId   社团id
     */
    void changeManage(String stuCode, Integer assId);

    /**
     * 通过登录系统管理人员信息查询此管理加入的所有能被他管理的社团，如果是超级管理，则返回所有社团信息
     *
     * @param stuCode 学号
     * @param role    角色
     * @return 返回数据
     */
    List<Association> findByStuCodeAndRole(String stuCode, String role);

    /**
     * 审核中心，查询所有社团信息
     *
     * @return 返回数据
     */
    List<AssoSimplify> findAll();

    /**
     * 查找此社团所有管理人员信息
     *
     * @param assId 社团id
     * @return 返回数据
     */
    List<Student> findAllManage(Integer assId);
}
